support/cas-server-support-ehcache3-ticket-registry/src/main/java/org/apereo/cas/config/Ehcache3TicketRegistryConfiguration.java

```diff
@@ -0,0 +1,205 @@
+                .resourcePool(ehcacheProperties.getResourcePoolName(), ehcacheProperties.getResourcePoolSize(), MemoryUnit.valueOf(ehcacheProperties.getResourcePoolUnits())));
+            return clusterConfigBuilder.build();
+        } else {
+            val rootDirectory = ehcacheProperties.getRootDirectory();
+            val rootDirectoryFile = new File(rootDirectory);
+            if (!rootDirectoryFile.exists()) {
+                LOGGER.debug("Creating folder for ehcache ticket registry disk cache [{}]", rootDirectory);
+                val mkdirResult = rootDirectoryFile.mkdirs();
+                if (!mkdirResult) {
+                    LOGGER.warn("Unable to create folder for ehcache ticket registry disk cache [{}]", rootDirectory);
+                }
+            }
+            return new DefaultPersistenceConfiguration(rootDirectoryFile);
+        }
+    }
+
+    @Bean
+    @ConditionalOnMissingBean
+    public javax.cache.CacheManager ehcache3TicketCacheManager(
+        @Qualifier ("ehcache3CacheManagerConfiguration") final ServiceCreationConfiguration ehcache3CacheManagerConfiguration) {
+        val ehcacheProperties = casProperties.getTicket().getRegistry().getEhcache3();
+        val cachingProvider = Caching.getCachingProvider(EhcacheCachingProvider.class.getName());
+        val ehcacheProvider = (EhcacheCachingProvider) cachingProvider;
+        val statisticsAllEnabled = ehcacheProperties.isEnableStatistics() ? ConfigurationElementState.ENABLED : ConfigurationElementState.DISABLED;
+        val managementAllAllEnabled = ehcacheProperties.isEnableManagement() ? ConfigurationElementState.ENABLED : ConfigurationElementState.DISABLED;
+        val jsr107Config = new Jsr107Configuration(null, new HashMap<>(),
+            false, managementAllAllEnabled, statisticsAllEnabled);
+        val configuration = new DefaultConfiguration(ehcacheProvider.getDefaultClassLoader(),
+            ehcache3CacheManagerConfiguration,
+            jsr107Config);
+        return ehcacheProvider.getCacheManager(ehcacheProvider.getDefaultURI(), configuration);
+    }
+
+    private CacheConfiguration<String, Ticket> buildCacheConfiguration(final TicketDefinition ticketDefinition) {
+        val ehcacheProperties = casProperties.getTicket().getRegistry().getEhcache3();
+        val terracottaClusterUri = ehcacheProperties.getTerracottaClusterUri();
+
+        CacheEventListenerConfigurationBuilder cacheEventListenerConfiguration = CacheEventListenerConfigurationBuilder
+            .newEventListenerConfiguration(new CasCacheEventListener(),
+                EventType.CREATED, EventType.UPDATED, EventType.EXPIRED, EventType.REMOVED, EventType.EVICTED)
+            .ordered().asynchronous();
+
+        val storageTimeout = ticketDefinition.getProperties().getStorageTimeout();
+        val expiryPolicy = ehcacheProperties.isEternal()
+            ?
+            ExpiryPolicyBuilder.noExpiration()
+            :
+            ExpiryPolicyBuilder.expiry()
```

Comment:
Nitpicking here; it would be easier to read this line if you merged it all in one line: `? xyz : abc `